cmd/fleet/serve.go

```diff
@@ -161,7 +161,10 @@ the way that the Fleet server works.
 			var ds fleet.Datastore
 			var carveStore fleet.CarveStore
 			var installerStore fleet.InstallerStore
-			mailService := mail.NewService()
+			mailService, err := mail.NewService(config)
+			if err != nil {
+				level.Error(logger).Log("err", err, "msg", "failed to configure mailing service")
```

Comment:
another candidate for https://github.com/fleetdm/fleet/issues/8935

not a blocker for approving, but I'm curious about why not do `initFatal` like we do for most things even if they're not required for basic server functions (like eg: carves)

Reply:
> another candidate for #8935
> 
> not a blocker for approving, but I'm curious about why not do `initFatal` like we do for most things even if they're not required for basic server functions (like eg: carves)

I think because technically you could misconfigure SMTP and it wouldn't cause Fleet to exit fatal. So I was trying my best to keep the existing behavior but also surface the issue to the admin.